jrnl/Journal.py

```diff
@@ -96,7 +96,7 @@ def _encrypt(self, text: str) -> bytes:
 
         return self.encryption_method.encrypt(text)
 
-    def open(self, filename=None):
+    def open(self, filename: str | None = None) -> "Journal":
```

Comment:
Can we use `Optional[str]` instead of `str | None`?

Reply:
They are equivalent right? So we could, but I have typed this consistently as `str | None` in #1614 
Anything wrong with the new syntax?